demo/implot_demo.jl

```diff
@@ -0,0 +1,1457 @@
+     end
+    #-------------------------------------------------------------------------
+    if CImGui.CollapsingHeader("Images")
+        CImGui.BulletText("Below we are displaying the font texture, which is the only texture we have\naccess to in this demo.")
+        CImGui.BulletText("Use the 'ImTextureID' type as storage to pass pointers or identifiers to your\nown texture data.")
+        CImGui.BulletText("See ImGui Wiki page 'Image Loading and Displaying Examples'.")
+        @cstatic(
+            bmin = ImVec2(0,0),
+            bmax = ImVec2(1,1),
+            uv0 = ImVec2(0,0),
+            uv1 = ImVec2(1,1),
+            tint = ImVec4(1,1,1,1),
+        begin
+            @c CImGui.SliderFloat2("Min", &bmin.x, -2, 2, "%.1f") #! exception = setfield! immutable struct of type ImVec2 cannot be changed
+            @c CImGui.SliderFloat2("Max", &bmax.x, -2, 2, "%.1f")
+            @c CImGui.SliderFloat2("UV0", &uv0.x, -2, 2, "%.1f")
+            @c CImGui.SliderFloat2("UV1", &uv1.x, -2, 2, "%.1f")
+            @c CImGui.ColorEdit4("Tint",&tint.x)
+            if ImPlot.BeginPlot("##image", "", "")
+                ImPlot.PlotImage("my image",CImGui.GetIO().Fonts.TexID, bmin, bmax, uv0, uv1, tint)
+                ImPlot.EndPlot()
+            end
+        end) # @cstatic
+    end
+    #-------------------------------------------------------------------------
+    if CImGui.CollapsingHeader("Realtime Plots")
+        CImGui.BulletText("Move your mouse to change the data!")
+        CImGui.BulletText("This example assumes 60 FPS. Higher FPS requires larger buffer size.")
+        @cstatic(
+            sdata1 = ScrollingBuffer(),
+            sdata2 = ScrollingBuffer(),
+            rdata1 = RollingBuffer(),
+            rdata2 = RollingBuffer(),
+            t = Float32(0),
+            history = Float32(10.0),
+            rt_axis = ImPlotAxisFlags_NoTickLabels,
+        begin
+            mouse = CImGui.GetMousePos() 
+            t += CImGui.GetIO().DeltaTime
+            AddPoint(sdata1, t, mouse.x * 0.0005)
+            AddPoint(rdata1, t, mouse.x * 0.0005)
+            AddPoint(sdata2, t, mouse.y * 0.0005)
+            AddPoint(rdata2, t, mouse.y * 0.0005)
+
+            @c CImGui.SliderFloat("History",&history,1,30,"%.1f s")
+            rdata1.span = history
+            rdata2.span = history
+            
+            ImPlot.SetNextPlotLimitsX(t - history, t, ImGuiCond_Always)
+            #? tried to use CircularBuffer here from DataStructures.jl, but it can't be properly passed into PlotLine functions...
+            if ImPlot.BeginPlot("##Scrolling", "", "", ImVec2(-1,150); 
+                flags = 0, #? IMPLOT_AUTO ?
+                x_flags = rt_axis, 
+                y_flags = rt_axis | ImPlotAxisFlags_LockMin
+            )
+                @c ImPlot.PlotShaded(&sdata1.data[1].x, &sdata1.data[1].y, 0; count = length(sdata1.data), offset = sdata1.offset, stride = 2 * sizeof(Float32), label_id = "Data 1")
```

Comment:
> For a first element itself I can write `pointer(sdata1.data, 1)`, but how then I shift it to the field offset?

`pointer(sdata1.data, 1) + fieldoffset(ImVec2, 2)`

Reply:
We can use pointer arithmetic if we want a absolute recreation of the demo, but I think something that's a native Julia structure demonstrating the same general concept and yielding equivalent output could suffice as well (and maybe even makes more sense). After all, we're trying to convince people we can do all the same things in Julia--not that Julia can be beaten into the style of arbitrary "dummy data" structures in a C++ demo. In all cases, don't fixate on this. Skip and we'll come back to it if necessary.

Reply:
Then maybe we can add some helper functions that can hide this ugly pointer arithmetic from user? 
Something like this:
```julia
function PlotLine(data::AbstractVector{T}, xfield::Symbol, yfield::Symbol; kwargs...) where T
    x_offset = fieldoffset(T, Base.fieldindex(T, xfield))
    y_offset = fieldoffset(T, Base.fieldindex(T, yfield))
    x_ptr = pointer(data, 1) + x_offset 
    y_ptr = pointer(data, 1) + y_offset 
    PlotLine(x_ptr, y_ptr, ...)
end
```

Reply:
```
struct ScrollingBuffer
    maxsize::Int
    offset::Int
    data::Vector{ImVec2}
end
```
FWIW, the `ScrollingBuffer` is not necessarily to be implemented in this way. I would write it like this:

```
struct ScrollingBuffer
    maxsize::Int
    x::Vector{Cfloat}
    y::Vector{Cfloat}
end
```

and just do some tweaks on the arguments:

`ImPlot.PlotShaded(sdata1.x, sdata1.y, 0; count = length(sdata1.x), offset = 0, stride = sizeof(Float32), label_id = "Data 1")`

Reply:
For this particular example - yes. (Or we can use StructArrays for the same purpose.)
Tо demonstrate generic approach for user-defined structures - probably no, since it will require that user change their data representation or write additional transformation or just because it is difficult to plot them as is.

Reply:
> For this particular example - yes. (Or we can use StructArrays for the same purpose.)
> Tо demonstrate generic approach for user-defined structures - probably no, since it will require that user change their data representation or write additional transformation just because it is difficult to plot them as is.

Thanks @sairus7 for the work on this, but I'm not sure we should create a whole new method to handle arrays of structs (instead we should do something like @Gnimuc's suggestion or do the necessary work in the demo itself). I can think about what to do with it. There might be an even simpler way. For now let's just move on and we'll come back to it.

Reply:
For now, I already have this merged and working. 

By the way, I am starting to think that adding kwarg `label_id` violates consistence with many others ImGui and ImPlot functions that have string id as their first argument.

Reply:
With one exception: its not working for vector of mutable structures...

Reply:
We can support both--I've been meaning to do this. Essentially, remove the import statements for each plot (so we shadow LibCImPlot) and provide a method without kwargs and all default values as in the C++ headers.